### DISCUSSION TITLE
I switched to Parcel to replace Webpack. Now fetching a json path no longer work. Why?

### DISCUSSION THREAD

**Discussion by turner** [parcel-bundler/parcel#7473] [Q&A]
I am moving from Webpack to Parcel to develop a web app. I have an array of json paths:

    const paths = [
        "resources/tracks/hg19_annotations.json",
        "resources/tracks/hg19_variants.json",
        "resources/tracks/hg19_encode.json",
        "resources/tracks/hg19_platinum_genomes.json",
        "resources/tracks/hg19_tutorials.json"
    ]

Each path is relative to the root of the app directory structure. I try to fetch one path as follows:

    let response
    try {
        response = await fetch(path)
    } catch (e) {
        console.error(e)
    }

    let json
    try {
        json = await response.json()
    } catch (e) {
        console.error(e)
    }

The code dies at `response.json()`. This code worked before I switched to Parcel. What am I missing? 

After further testing I see that no runtime loading of any asset of any kind in my project tree - png, jpeg, txt, csv, etc. - works in Parcel. All of this code has worked perfectly prior to switching to Parcel. What am I missing here?

**Comment by devongovett** [comment]
Take a look at this https://parceljs.org/languages/javascript/#url-dependencies

**Reply by turner** [reply]
> 

Yes, I found that doc. Here is what I am seeing:


    const path = new URL('resources/tracks/hg19_annotations.json', import.meta.url);
Using this approach transforms: path `resources/tracks/hg19_annotations.json` into `http://localhost:1234/hg19_annotations.80b994d9.js?1640022990317`. This is a JS file, not a JSON file. This is not correct.

If I use this in a test, the result is as before, an unfound path. The error message in console.log is a list of the source code of some unknown JS. Huh?

```
    let response
    try {
        response = await fetch(path)
    } catch (e) {
        AlertSingleton.present(e.message)
        console.error(e)
    }

    let txt
    try {
        txt = await response.text()
    } catch (e) {
        AlertSingleton.present(e.message)
        console.error(e)
    }

    console.log(`Track Registry retrieval ${ txt }`)


```
What is going on here? Why is my code be transformed like this and why is the result non functional code? None of this makes any sense.

**Reply by devongovett** [reply]
Transforming JSON into JS is indeed an issue. It's a little complex because sometimes we do want JSON to be inlined into JS, but not in this case.

As a workaround, you can try something like this in your `.parcelrc`:

```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.json": ["@parcel/transformer-raw"]
  }
}
```

That will override the default transformers for JSON. However, if you do need to load any JSON files synchronously (i.e. via `import`), then it will be a bit more complex as you need some way to distinguish between these cases. To solve that, you could make use of [named pipelines](https://parceljs.org/features/plugins/#named-pipelines).

```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "raw:*.json": ["@parcel/transformer-raw"]
  }
}
```

Now you'll get a JS module by default, but if you use the `raw:` scheme, you'll get a raw JSON file.

```js
new URL('raw:some.json', import.meta.url)
```

Sorry for the inconvenience. This is something we still need to figure out how to handle automatically.

**Reply by turner** [reply]
Cool. Thanks for the assist.
